src/NHibernate/Cfg/Environment.cs

```diff
@@ -381,6 +391,16 @@ public static IBytecodeProvider BytecodeProvider
 			set { BytecodeProviderInstance = value; }
 		}
 
+		/// <summary>
+		/// NHibernate's object instantiator.
+		/// </summary>
+		/// <remarks>
+		/// This should only be set before a configuration object
+		/// is created, otherwise the change may not take effect.
+		/// For entities see <see cref="IReflectionOptimizer"/> and its implementations.
+		/// </remarks>
+		public static IObjectsFactory ObjectsFactory { get; set; } = new ActivatorObjectsFactory();
```

Comment:
To avoid a possible breaking change, we should default to using `BytecodeProvider.ObjectsFactory` here. But for completeness, we should also use the objects factory to create the byte-code provider...

I may add a commit sorting this out.